app/models/retirement_manager.rb

```diff
@@ -0,0 +1,12 @@
+class RetirementManager
+  def self.check
+    ems_ids = MiqServer.my_server.zone.ext_management_system_ids
+    [LoadBalancer, OrchestrationStack, Vm, Service].each do |model|
+      table = model.arel_table
+      arel = table[:retires_on].not_eq(nil).or(table[:retired].eq(true))
+      arel = arel.and(table[:ems_id].in(ems_ids)) if model.column_names.include?('ems_id') # Service not assigned to ems_ids
+      items = model.where(arel)
+      items.each(&:retirement_check)
```

Comment:
@gmcculloug This PR handles my changes except checking if retired is true.

The retirement process used to set retired = true before entering the state machine, so looking for retired = true was correct. Now, we set retired = true at the end of the state machine, so we should only perform a retirement check on objects where retired != true.

Can we change the query to:  
table[:retires_on].not_eq(nil).and(table[:retired].not_eq(true))